### ISSUE TITLE
Recently introduced different behaviour of da.interp() when using floats vs DataArrays with new dim

### ISSUE THREAD

**Issue by veenstrajelmer** [pydata/xarray#7701]
### What happened?

In recent xarray versions, `da.interp()` behaves differently when using floats as input, in comparison to using a DataArray with a new dimension as input. In the attached example, the variables `interp_with_floats` and `interp_with_da` return a different amount of float/nan values:
```
[35.824154 35.810417 35.796684       nan       nan]
[35.824154 35.810417       nan       nan       nan]
```
This is the case with xarray 2023.1.0/2023.2.0, but was not yet happening in xarray 2022.6.0/2022.12.1.dev15+gd6d24507. I cannot find .interp() related changes in https://docs.xarray.dev/en/stable/whats-new.html. A CMEMS use case is described in https://github.com/Deltares/dfm_tools/issues/287

### What did you expect to happen?

Equal interpolation results with both methods since the input values are equal.

### Minimal Complete Verifiable Example

```Python
import numpy as np
import xarray as xr

ds = xr.Dataset()
so_np = np.array([[[35.819576, 35.82568 , 35.82873 ],
                   [35.819576, 35.824154, 35.831783],
                   [35.822628, 35.824154, 35.82873 ]],
                  
                  [[35.802788, 35.80584 , 35.815   ],
                   [35.815   , 35.810417, 35.821102],
                   [35.824154, 35.813473, 35.81805 ]],
                  
                  [[35.786003, 35.789055,       np.nan],
                   [35.807365, 35.796684,       np.nan],
                   [35.824154, 35.80584 ,       np.nan]],
                  
                  [[35.776848,       np.nan,       np.nan],
                   [35.792107,       np.nan,       np.nan],
                   [35.822628,       np.nan,       np.nan]],
                  
                  [[35.781425,       np.nan,       np.nan],
                   [35.792107,       np.nan,       np.nan],
                   [35.789055,       np.nan,       np.nan]]])
ds['so'] = xr.DataArray(so_np,dims=('depth','latitude','longitude'))
ds['longitude'] = xr.DataArray([-9.6, -9.5, -9.4], dims=('longitude'))
ds['latitude'] = xr.DataArray([42.9, 43.0, 43.1], dims=('latitude'))

x_xr = xr.DataArray([-9.5],dims=('plipoints'))
y_xr = xr.DataArray([43],dims=('plipoints'))


interp_with_floats = ds.interp(longitude=x_xr[0], latitude=y_xr[0], method='linear').so #selecting one value from the da drops the new plipoints dimension
interp_with_da_existing = ds.interp(longitude=x_xr.values, latitude=y_xr.values, method='linear').so.isel(longitude=0,latitude=0) #using the DataArray values keeps lat/lon dimenions, gives the same interp result
interp_with_da_newdim = ds.interp(longitude=x_xr, latitude=y_xr, method='linear').so.isel(plipoints=0) #using the DataArray introduces a plipoints dimension, which gives different interp result
print(interp_with_floats.to_numpy())
print(interp_with_da_existing.to_numpy())
print(interp_with_da_newdim.to_numpy())
print(xr.__version__)

assert (interp_with_floats.isnull()==interp_with_da_existing.isnull()).all() #success
assert (interp_with_floats.isnull()==interp_with_da_newdim.isnull()).all() #fails with scipy>=1.10.0
```

### MVCE confirmation

- [X] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [X] Complete example — the example is self-contained, including all data and the text of any traceback.
- [X] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [X] New issue — a search of GitHub Issues suggests this is not a duplicate.

### Environment

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:59:45) [MSC v.1929 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 85 Stepping 0, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: ('Dutch_Netherlands', '1252')
libhdf5: 1.12.2
libnetcdf: 4.8.1

xarray: 2023.1.0
pandas: 1.5.3
numpy: 1.23.5
scipy: 1.10.1
netCDF4: 1.6.2
pydap: installed
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.3.3
cfgrib: None
iris: None
bottleneck: 1.3.6
dask: 2022.9.2
distributed: 2022.9.2
matplotlib: 3.6.1
cartopy: 0.21.0
seaborn: 0.10.1
numbagg: None
fsspec: 2022.10.0
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 65.5.0
pip: 23.0.1
conda: None
pytest: 7.1.3
mypy: None
IPython: 7.33.0
sphinx: 5.3.0
</details>

**Comment by headtr1ck** [comment]
How did you test this? I just tried it with 2023.03.0 and 2022.06.0 with all other packages the same and I get the same results (the one you say is new).

**Comment by veenstrajelmer** [comment]
Hmm, that is interesting and helpful. I have tested in on a laptop with an older environment. I assumed the xarray version was the difference, but I guess there is something else that is causing it if you cannot reproduce it.

### Environment where it does work as expected

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:59:45) [MSC v.1929 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: ('Dutch_Netherlands', '1252')
libhdf5: 1.12.2
libnetcdf: 4.8.1

xarray: 2022.6.0
pandas: 1.5.0
numpy: 1.23.3
scipy: 1.9.1
netCDF4: 1.6.1
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.3.2
cfgrib: None
iris: None
bottleneck: 1.3.5
dask: 2022.02.1
distributed: 2022.2.1
matplotlib: 3.6.0
cartopy: 0.21.0
seaborn: None
numbagg: None
fsspec: 2022.8.2
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 65.3.0
pip: 22.2.2
conda: None
pytest: None
IPython: 7.33.0
sphinx: 5.2.1
</details>

**Comment by veenstrajelmer** [comment]
I just discovered that it is not per se a difference between floats/da, but it has to do with the creation of the new dimension (`plipoints` in this case):

```
interp_with_floats = ds.interp(longitude=x_xr[0], latitude=y_xr[0], method='linear').so #selecting one value drops the new plipoints dimension
interp_with_da_existing = ds.interp(longitude=x_xr.values, latitude=y_xr.values, method='linear').so.isel(longitude=0,latitude=0) #using the DataArray values keeps lat/lon dimenions, gives the same interp result
interp_with_da_newdim = ds.interp(longitude=x_xr, latitude=y_xr, method='linear').so.isel(plipoints=0) #using the DataArray introduces a plipoints dimension, which gives different interp result
print(interp_with_floats.to_numpy())
print(interp_with_da_existing.to_numpy())
print(interp_with_da_newdim.to_numpy())
print(xr.__version__)

assert (interp_with_floats.isnull()==interp_with_da_existing.isnull()).all() #success
assert (interp_with_floats.isnull()==interp_with_da_newdim.isnull()).all() #fails
```

**Comment by veenstrajelmer** [comment]
@headtr1ck I now discovered it might be due to using interpn, I can at least reproduce it with scipy only: https://github.com/scipy/scipy/issues/19318

However, when looking at scipy only, the version does not matter. When doing it from xarray, the scipy version does make a difference. Could it be that xarray moved from `interp1d` to `interpn` in this case?

**Comment by veenstrajelmer** [comment]
@headtr1ck: There were new insights from the scipy issue. I would say the shown behaviour is expected and this issue can be closed. I tried several alternatives for getting a larger coverage from the interpolation that I do (less nans). I tried with interp1d in a loop and with a lat/lon grid of which I took the diagonal. This is killing for performance.

**For others who bump into this problem**
In the end, I went with a combination of linear and neareast interpolation. This increases the coverage to a maximum, but still does not return values in locations where nans are present. So this is my ideal case. The impact on the performance is not noticeable. An example is given in the original dfm_tools issue: https://github.com/Deltares/dfm_tools/issues/287#issuecomment-1746938349
